chainstate/test-framework/src/block_builder.rs

```diff
@@ -112,6 +112,19 @@ impl<'f> BlockBuilder<'f> {
         self
     }
 
+    /// Returns regular transaction output(s) if any, otherwise returns block reward outputs
+    fn get_regular_block_outputs(outputs: BlockOutputs) -> BlockOutputs {
```

Comment:
Comment describes logic that is more complex then a function name suggests. Maybe if we need to distinguish outputs from reward and tx we should make `BlockOutputs` type a struct with 2 dedicated fields instead of mixing them in one collection? That would be more explicit. Does it make sense?

Reply:
looking again at this function I still have a filling that something should be done about it. Because it's kind of misleading: not only regular outputs can be returned but also reward. What if we rename it to `filter_outputs`?